src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc

```diff
@@ -0,0 +1,93 @@
+= Hawkular Data Mining - Predictive Charts
+Pavol Loffay
+2016-1-20
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, data mining
+
+I would like to present you functionality of Hawkular Data Mining module.
+
+Hawkular Data Mining module can be deployed in Hawkular and predict all metrics. Predicted values are
+automatically sent to Alerts where conditions are evaluated and an alert can be triggered.
+
+=== Time series models
+At the beginning we link:http://www.hawkular.org/blog/2015/10/24/hawkular-datamining.html[started] with linear
+regression in Apache Spark. This simple model was a good start. Now we use Holt's linear trend method which is
+a variant of exponential smoothing with linear trend. For forecasting an alert is important to model a trend. ARIMA models
+were also proposed, however they require stationary time series and periodical recalculation of parameters would
+have to be done. Which requires all historical data in place. For seasonal time series we plan to sue Holt Winters
+model which is just an extension of the model which is currently used.
+
+=== Data Mining with Hawkular
+Predictions are by default disabled. Enabling can by done by creating Relationship from Tenant to Metric, MetricType or Tenant.
+Relationship to Tenant enables forecasting of all metrics under given Tenant, to MetricType enables all metrics of
+given type. Lower levels override highers.
+
+Configuration has only one property -- interval in seconds. If a new metric data is available
+module predicts future values up to last timestamp + prediction interval. Then predicted values are sent to Alerts where
+conditions are evaluated.
+
+.Hawkular with integrated Data Mining:
+--
+ $ git clone -b datamining git@github.com:hawkular/hawkular.git
+ $ cd hawkular
+ $ mvn clean install -Pdev
+ $ dist/target/hawkular-*/bin/standalone.sh
+--
+
+.Enable predictions for all metrics of given tenant.
+--
+ $ tenant=$(curl -s GET 'http://jdoe:password@localhost:8080/hawkular/inventory/tenant'| grep --color=never -oP 'path" : "\K/t;[0-9a-z\-]+')
+ $ curl -ivX POST -H "Content-Type: application/json" 'http://jdoe:password@localhost:8080/hawkular/inventory/tenants/relationships' -d '{
+     "name": "__inPrediction",
+     "source": "'$tenant'",
+     "target": "'$tenant'",
+     "properties": {"predictionInterval": 150}
+ }'
```

Comment:
TODO I will rename it to forecastingHorizon as this collide with confidence interval which is in time series theory called  also prediction interval...